browser/lib/util/http.ts

```diff
@@ -0,0 +1,176 @@
+import * as Utils from '../../../common/lib/util/utils';
+import Defaults from '../../../common/lib/util/defaults';
+import ErrorInfo from '../../../common/lib/types/errorinfo';
+import { IHttp, RequestCallback, RequestParams } from '../../../common/types/http';
+import HttpMethods from '../../../common/constants/HttpMethods';
+
+type Rest = any;
+type Realtime = any;
+
+const noop = function() {};
+
+function shouldFallback(err: ErrorInfo) {
```

Comment:
It's me again, on the top of abbreviations. 😁 

For me, `err` is a confusing abbreviation to use here. I can see from the type information you've added that it's clearly not a JS-built-in [Error](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/error), but that would have been my default assumption. I would expand fully to `errorInfo` (and, yes, I accept that this would sub-optimally be `errorInfo` rather than `errorInformation` - however I accept that this is the state of play for this type as `ErrorInfo` is [baked into the spec.](https://docs.ably.com/client-lib-development-guide/features/#TI1)).